util/plugins/eu.esdihumboldt.util.test/src/eu/esdihumboldt/util/geometry/InterpolationTest.java

```diff
@@ -0,0 +1,209 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Envelope;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.LineString;
+
+import eu.esdihumboldt.util.geometry.interpolation.ArcInterpolation;
+import eu.esdihumboldt.util.geometry.interpolation.Interpolation;
+
+/**
+ * Test for the Interpolation of arc algorithm
+ * 
+ * @author Arun
+ */
+@RunWith(Parameterized.class)
+public class InterpolationTest {
+
+	private final int testIndex;
+	private final Coordinate[] arcCoordinates;
+	@SuppressWarnings("rawtypes")
+	private final Class generatedGeometryType;
+
+	private static final int MAX_SIZE = 200;
+	private static final double e = 0.025;
+
+	private static final boolean DRAW_IMAGE = false;
+
+	/**
+	 * Constructor for parameterized test
+	 * 
+	 * @param testIndex Index of test parameters
+	 * @param coordinates input arc coordinates
+	 * @param geometry type of output geometry
+	 */
+	@SuppressWarnings("rawtypes")
+	public InterpolationTest(int testIndex, Coordinate[] coordinates, Class geometry) {
+		this.testIndex = testIndex;
+		this.arcCoordinates = coordinates;
+		this.generatedGeometryType = geometry;
+	}
+
+	/**
+	 * Passing arc geometries
+	 * 
+	 * @return Collection of arc coordinates and type of generated geometry
+	 */
+
+	@SuppressWarnings("rawtypes")
+	@Parameters
+	public static Collection addCoordiantes() {
+		return Arrays.asList(new Object[][] { //
+				{ 0, new Coordinate[] { new Coordinate(577869.169, 5917253.678),
+						new Coordinate(577871.772, 5917250.386),
+						new Coordinate(577874.884, 5917253.202) }, //
+						LineString.class }, //
+				{ 1, new Coordinate[] { new Coordinate(577738.2, 5917351.786),
+						new Coordinate(577740.608, 5917347.876),
+						new Coordinate(577745.185, 5917348.135) }, //
+						LineString.class }, //
+				{ 2, new Coordinate[] { new Coordinate(240, 280), new Coordinate(210, 150),
+						new Coordinate(300, 100) }, //
+						LineString.class }, //
+				{ 3, new Coordinate[] { new Coordinate(8, 8), new Coordinate(12, 16),
+						new Coordinate(16, 8) }, //
+						LineString.class }, //
+				{ 4, new Coordinate[] { new Coordinate(8, 8), new Coordinate(12, 6.5),
+						new Coordinate(16, 8) }, //
+						LineString.class }, //
+				{ 5, new Coordinate[] { new Coordinate(3, 10.5), new Coordinate(4, 7.75),
+						new Coordinate(8, 8) }, //
+						LineString.class } //
+		});
+	}
+
+	/**
+	 * test algorithm
+	 */
+	@Test
+	public void testInterpolation() {
+		System.out.println("-- Test-" + testIndex + " begin --");
+
+		Interpolation<LineString> interpolation = new ArcInterpolation(this.arcCoordinates, e);
+		Geometry interpolatedArc = interpolation.interpolateRawGeometry();
+
+		assertNotNull(interpolatedArc);
+		Assert.assertEquals(interpolatedArc.getClass(), generatedGeometryType);
```

Comment:
It would be good if the test would test some of the characteristics of the interpolated arcs.
For instance that the positional error of the points is indeed equal to or less than the setting and maybe if the points lie on the grid.